crux/src/Crux.hs

```diff
@@ -213,7 +214,8 @@ mkOutputConfig ::
   (msgs -> SayWhat) -> Maybe CruxOptions ->
   OutputConfig msgs
 mkOutputConfig withColors outHandle errHandle logMessageToSayWhat opts =
-  let lgWhat = let la = LJ.LogAction $ logToStd withColors outHandle errHandle
+  let withColorsConsensus = withColors && not (maybe False noColors opts)
+      lgWhat = let la = LJ.LogAction $ logToStd withColorsConsensus outHandle errHandle
```

Comment:
What if we dropped the `withColors` argument here? Callers can use the opts to set an override if  they want it.

For the default, we already have System.Console.ANSI available, so we could use `hSupportsANSIColor` to get the default value, at the cost of changing `mkOutputConfig` from a pure function to an IO monad operation.  This will have a little bit of impact, but the removal of the extra argument and the possibility that this will automatically default to doing the right thing could be pretty helpful.  What do you think?

Reply:
I actually tried that initially, but then I wasn't sure where to force the value in places such as:

https://github.com/GaloisInc/crucible/blob/ede38ab268f7eed2264b7276b06dfa3e6c6670e5/uc-crux-llvm/src/UCCrux/LLVM/Main.hs#L90-L93

but I guess what you'd do is more or less:
```haskell
mainWithOutputTo h =
  mainWithOutputConfig $
    set noColors True
    >>> Crux.mkOutputConfig False h h ucCruxLLVMLoggingToSayWhat
```
?

Basically I just want to make sure that we allow:
- a sensible default (e.g. what you suggest of detecting ANSI support),
- a way for a tool override the default for its main use case,
- a way for users to refuse colors.

One thing that I'm not sure about is whether the consensus should be colored output unless anyone disagrees, or whether, say, command-line options preference should prevail over tool's default preference.

We might want to rephrase the option/doc depending on the semantics we land upon.

Reply:
It's actually slightly less nice, because `mkOutputConfig` receives a `Maybe CruxOptions`, and so in the `Nothing` case, we can't set `noColors` to `True` short of creating a whole default `CruxOptions`...

Reply:
I was thinking of the `set noColors true` technique for the forcing locations, but based on your latest observation, maybe it's better to keep the `Bool` but update `defaultOutputConfig` in Crux to use the `hSupportsANSIColor` in the IO monad.  That still allows for the "normal" config, but could be less invasive.  Does that seem reasonable?

To your semantics question, I'd assume that the `noColors` set to`True` would override any default coloring, regardless of origin (cfgfile, env-var, or cmd-line).

Reply:
Is `hSupportsANSIColor` portable to Windows? That would be my only concern.  Otherwise, it seems nice to query it when making a `defaultOutputConfig`.

Reply:
The documentation isn't specific on this point, but I imagine this is a variation of the `hSupportsANSI`, which is reported to work for Windows.  These functions _are_ GHC specific, but I think we're already pretty far down that particular path.

Reply:
Updated. Currently the option is set if `stdout` supports ANSI. Should it instead be the conjunction of support on `stdout` and `stderr`, or does this seem better?

Reply:
Good point.  I suspect it would be best to keep those independently indicated down to the point where the output is configured and then configure each output according to the corresponding channel's color capabilities.  Otherwise one or the other will be suboptimal if the compatibilities don't match.

Reply:
Alright, so I rebased over the current master branch, with options to independently disable colors for output, errors, or both.